validator/client/propose.go

```diff
@@ -231,20 +383,48 @@ func (v *validator) signBlock(ctx context.Context, pubKey [48]byte, epoch types.
 	}
 
 	var sig bls.Signature
-	blockRoot, err := helpers.ComputeSigningRoot(b, domain.SignatureDomain)
-	if err != nil {
-		return nil, nil, errors.Wrap(err, signingRootErr)
-	}
-	sig, err = v.keyManager.Sign(ctx, &validatorpb.SignRequest{
-		PublicKey:       pubKey[:],
-		SigningRoot:     blockRoot[:],
-		SignatureDomain: domain.SignatureDomain,
-		Object:          &validatorpb.SignRequest_Block{Block: b},
-	})
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "could not sign block proposal")
+	switch b.Version() {
+	case version.Altair:
+		block, ok := b.Proto().(*ethpb.BeaconBlockAltair)
+		if !ok {
+			return nil, nil, errors.New("could not convert obj to beacon block altair")
+		}
+		blockRoot, err := helpers.ComputeSigningRoot(block, domain.SignatureDomain)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, signingRootErr)
+		}
+		sig, err = v.keyManager.Sign(ctx, &validatorpb.SignRequest{
+			PublicKey:       pubKey[:],
+			SigningRoot:     blockRoot[:],
+			SignatureDomain: domain.SignatureDomain,
+			Object:          &validatorpb.SignRequest_BlockV2{BlockV2: block},
+		})
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "could not sign block proposal")
+		}
+		return sig.Marshal(), domain, nil
```

Comment:
This stuff is duplicated below except for the Object: field of SignRequest

Reply:
I think refactoring is an overkill and turning it for worse. The only functions to refactor out `ComputeSigningRoot` and `Sign`. It will look like below and I'm not sure how to express `isSignRequest_Object` as argument

```go
func (v *validator) blockSignature(ctx  context.Context, 
	obj ssz.HashRoot, 
	domain []byte,
	pubKey [48]byte,
	blk *validatorpb.SignRequest_Block) ([]byte, error) {
	blockRoot, err := helpers.ComputeSigningRoot(obj, domain)
	if err != nil {
		return nil, errors.Wrap(err, signingRootErr)
	}
	sig, err := v.keyManager.Sign(ctx, &validatorpb.SignRequest{
		PublicKey:       pubKey[:],
		SigningRoot:     blockRoot[:],
		SignatureDomain: domain,
		Object:          blk,
	})
	return sig.Marshal(), err
}
```